data/crac/crac-impl/src/main/java/com/powsybl/openrao/data/cracimpl/AbstractSingleNetworkElementActionAdderImpl.java

```diff
@@ -50,4 +56,8 @@ public NetworkActionAdder add() {
     protected abstract void assertSpecificAttributes();
 
     protected abstract String getActionName();
+
+    protected String createActionName(Object specificAttribute) {
+        return id == null ? String.format("%s_%s_%s", getActionName(), networkElementId, specificAttribute) : id;
```

Comment:
rename getActionName to getActionTypeName to avoid confusions between these two methods.